packages/go_router/lib/src/parser.dart

```diff
@@ -104,7 +104,7 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
     }
     if (configuration.matches.last is ImperativeRouteMatch) {
       configuration =
-          (configuration.matches.last as ImperativeRouteMatch).matches;
+          (configuration.matches.last as ImperativeRouteMatch<dynamic>).matches;
```

Comment:
can this be `Object?` ?

Reply:
we should avoid using dynamic as much as possible unless you are handling json.

Reply:
Done